Bottom: 3bb6b11e659c7a8851931ea7585f40f77d21f921
Top:    f72e0350d36f1174648f8089a6e78df72eaa7a3d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-19 17:44:31 +0100

Refresh of attempt-some-gl-optimisation

---

diff --git a/src/hid/common/actions.c b/src/hid/common/actions.c
index 0256772..8678ad9 100644
--- a/src/hid/common/actions.c
+++ b/src/hid/common/actions.c
@@ -180,7 +180,7 @@ hid_find_action (const char *name, void **context)
 
   for (i = 0; i < n_actions; i++)
     if (strcasecmp (all_actions[i].action.name, name) == 0) {
-      if (*context != NULL)
+      if (context != NULL)
         *context = all_actions[i].context;
       return &(all_actions[i].action);
     }
@@ -358,7 +358,8 @@ another:
   while (*sp && *sp != '(')
     *cp++ = *sp++;
   *cp++ = 0;
-  sp++;
+  if (*sp)
+    sp++;
 
   /*
    * we didn't find a leading ( so invoke the action
